src/index.js

```diff
@@ -130,7 +130,15 @@ export default async function fetch(url, options_) {
 				const location = headers.get('Location');
 
 				// HTTP fetch step 5.3
-				const locationURL = location === null ? null : new URL(location, request.url);
+				let locationURL;
+				try {
+					locationURL = location === null ? null : new URL(location, request.url);
+				} catch (error) {
+					// error can only be invalid URL
+					reject(new FetchError(`uri requested responds with an invalid redirect URL: ${location}`, 'invalid-redirect'));
```

Comment:
Have we looked at what error browsers gives back here? 🤔

Reply:
Chrome:

<img width="463" alt="Screenshot 2021-11-18 at 16 37 18" src="https://user-images.githubusercontent.com/189580/142446679-0502fe34-7d4b-4882-a081-8ec2884b198f.png">

Safari:

<img width="1076" alt="Screenshot 2021-11-18 at 16 37 43" src="https://user-images.githubusercontent.com/189580/142446761-0cde4d03-753a-4034-83fb-d6a6864b0675.png">

Right, browsers doesn't give any more info on errors 😅

Reply:
Note of caution, Had some errors in the past where we don't return... 

```diff
- reject(...)
+ return reject(...)
```

Reply:
Thinking if we should throw a `TypeError` here as well to match what the browser gives back... but obvious with a better error message.

`new TypeError('Failed to fetch; Redirect url is not valid, Got: ' + location)`

Reply:
In that case I think it's true for every `FetchError`, no? 🤔

Reply:
> In that case I think it's true for every `FetchError`, no? 🤔

pretty much 😅

Reply:
From what I can tell, in node-fetch TypeErrors are mostly "programmer error" and FetchErrors are "remote/response error" 

Some TypeErrors:

node-fetch cannot load ${url}. URL scheme "${parsedURL.protocol.replace(/:$/, '')}" is not supported.');
Redirect option '${request.redirect}' is not a valid value of RequestRedirect'));
body used already for: ${data.url}');
Invalid referrerPolicy: ${referrerPolicy}');
Invalid referrerPolicy: ${policy}');
${parsedURL} is an url with embedded credentails.');
Request with GET/HEAD method cannot have body');
Expected signal to be an instanceof AbortSignal or EventTarget');
Header name must be a valid HTTP token [${name}]');
Invalid character in header content ["${name}"]');
Header pairs must be iterable');
Each header pair must be an iterable object');
Each header pair must be a name/value tuple');
Failed to construct \'Headers\': The provided value is not of type \'(sequence<sequence<ByteString>> or record<ByteString, ByteString>)');
Failed to fetch');
no or bad content-type header, no multipart boundary');

Some FetchErrors:

**Invalid response body while trying to fetch ${data.url}: ${error.message}**
Invalid response body while trying to fetch ${this.url}: ${error_.message}
content size at ${data.url} over limit: ${data.size}
Could not create Buffer from response body for ${data.url}: ${error.message}
Premature close of server response while trying to fetch ${data.url}

Whereas in the browser there only seem to be [AbortError and TypeErrors](https://developer.mozilla.org/en-US/docs/Web/API/fetch#exceptions)

Reply:
I agree, but I think that we should fix that in a separate pull request? I think that this one is ready for merging as soon as the linting error is fixed?